src/Selenium2Driver.php

```diff
@@ -560,58 +518,33 @@ public function getAttribute($xpath, $name)
     public function getValue($xpath)
     {
         $element = $this->findElement($xpath);
-        $elementName = strtolower($element->name());
-        $elementType = strtolower($element->attribute('type'));
+        $elementName = strtolower($element->getTagName());
+        $elementType = strtolower($element->getAttribute('type'));
 
         // Getting the value of a checkbox returns its value if selected.
         if ('input' === $elementName && 'checkbox' === $elementType) {
-            return $element->selected() ? $element->attribute('value') : null;
+            return $element->isSelected() ? $element->getAttribute('value') : null;
         }
 
         if ('input' === $elementName && 'radio' === $elementType) {
-            $script = <<<JS
-var node = {{ELEMENT}},
-    value = null;
-
-var name = node.getAttribute('name');
-if (name) {
-    var fields = window.document.getElementsByName(name),
-        i, l = fields.length;
-    for (i = 0; i < l; i++) {
-        var field = fields.item(i);
-        if (field.form === node.form && field.checked) {
-            value = field.value;
-            break;
-        }
-    }
-}
-
-return value;
-JS;
-
-            return $this->executeJsOnElement($element, $script);
+            $element = new WebDriverRadios($element);
```

Comment:
I would not override the `$element` variable. `WebDriverRadios` is not the element, but a different helper.

Reply:
👍